examples/cis2-wccd-upgrade/src/lib.rs

```diff
@@ -0,0 +1,2030 @@
+                self.token.get_mut(from).ok_or(ContractError::InsufficientFunds)?;
+            ensure!(from_state.balance >= amount, ContractError::InsufficientFunds);
+            from_state.balance -= amount;
+        }
+        let mut to_state = self.token.entry(*to).or_insert_with(|| AddressState {
+            balance:   0u64.into(),
+            operators: state_builder.new_set(),
+        });
+        to_state.balance += amount;
+
+        Ok(())
+    }
+
+    /// Update the state adding a new operator for a given token id and address.
+    /// Results in an error if the token id does not exist in the state.
+    /// Succeeds even if the `operator` is already an operator for this
+    /// `token_id` and `address`.
+    fn add_operator(
+        &mut self,
+        owner: &Address,
+        operator: &Address,
+        state_builder: &mut StateBuilder<S>,
+    ) -> ContractResult<()> {
+        let mut owner_state = self.token.entry(*owner).or_insert_with(|| AddressState {
+            balance:   0u64.into(),
+            operators: state_builder.new_set(),
+        });
+        owner_state.operators.insert(*operator);
+        Ok(())
+    }
+
+    /// Update the state removing an operator for a given token id and address.
+    /// Results in an error if the token id does not exist in the state.
+    /// Succeeds even if the `operator` is not an operator for this `token_id`
+    /// and `address`.
+    fn remove_operator(&mut self, owner: &Address, operator: &Address) -> ContractResult<()> {
+        self.token.entry(*owner).and_modify(|address_state| {
+            address_state.operators.remove(operator);
+        });
+        Ok(())
+    }
+
+    fn mint(
+        &mut self,
+        token_id: &ContractTokenId,
+        amount: ContractTokenAmount,
+        owner: &Address,
+        state_builder: &mut StateBuilder<S>,
+    ) -> ContractResult<()> {
+        ensure_eq!(token_id, &TOKEN_ID_WCCD, ContractError::InvalidTokenId);
+        let mut owner_state = self.token.entry(*owner).or_insert_with(|| AddressState {
+            balance:   0u64.into(),
+            operators: state_builder.new_set(),
+        });
+
+        owner_state.balance += amount;
+        Ok(())
+    }
+
+    fn burn(
+        &mut self,
+        token_id: &ContractTokenId,
+        amount: ContractTokenAmount,
+        owner: &Address,
+    ) -> ContractResult<()> {
+        ensure_eq!(token_id, &TOKEN_ID_WCCD, ContractError::InvalidTokenId);
+        if amount == 0u64.into() {
+            return Ok(());
+        }
+
+        let mut from_state = self.token.get_mut(owner).ok_or(ContractError::InsufficientFunds)?;
+        ensure!(from_state.balance >= amount, ContractError::InsufficientFunds);
+        from_state.balance -= amount;
+
+        Ok(())
+    }
+}
+
+// Contract functions
+
+/// Initialize the state contract instance with no initial tokens.
+/// Logs a `Mint` event for the single token id with no amounts.
+/// TODO: move event logs to proxy
```

Comment:
I'm not sure about the status of the todos in this PR.

Is this something you want to handle now, or is it deliberately set for later?